### ISSUE TITLE
BUG/TST: resid_deviance, GEE, GLM, family

### ISSUE THREAD

**Issue by josef-pkt** [statsmodels/statsmodels#4209]
https://github.com/statsmodels/statsmodels/issues/4208#issuecomment-357811452

https://github.com/statsmodels/statsmodels/commit/6b8a24b68f8b1ade2dcba7740d7f96a4e2ca6739#diff-20016798d17fceba2ff1d3dfec9f2e61
has most likely introduced a bug in GEEResults.resid_deviance by changing the behavior of the family resid method. GLMResults has a compensating change but not GEEResults.

AFAIR, there are no unit tests for GEEResults resid_xxx. Those could be compared to the GLM counterparts by using independence covariance structure.

families don't have unit tests by themselves

Needs checking before 0.9

**Comment by josef-pkt** [comment]
Bug for resid_deviance in GEE confirmed, other resid are fine:

```
In [14]:
for resid in ['resid_response', 'resid_pearson', 'resid_deviance', 'resid_working', 'resid_anscombe']:
    r1 = getattr(rslt1, resid)
    r2 = getattr(rslt2, resid)
    print(resid, np.max(np.abs(r1 - r2)))

resid_response 1.88737914186e-14
resid_pearson 4.46309655899e-14
resid_deviance 2.81714847623
resid_working 1.31894495325e-13
resid_anscombe 3.60822483003e-14
```

based on test case in test_gee.py
```
    #def test_compare_logit(self):

vs = Independence()
family = families.Binomial()

Y = 1 * (np.random.normal(size=100) < 0)
X1 = np.random.normal(size=100)
X2 = np.random.normal(size=100)
X3 = np.random.normal(size=100)
groups = np.random.randint(0, 4, size=100)

D = pd.DataFrame({"Y": Y, "X1": X1, "X2": X2, "X3": X3})

mod1 = GEE.from_formula("Y ~ X1 + X2 + X3", groups, D,
                        family=family, cov_struct=vs)
rslt1 = mod1.fit()

mod2 = GLM.from_formula("Y ~ X1 + X2 + X3", data=D, family=family)
rslt2 = mod2.fit(disp=False)

assert_almost_equal(rslt1.params.values, rslt2.params.values,
                    decimal=10)
```

**Comment by josef-pkt** [comment]
suggested fix as mentioned in comment in #4208, which still requires minimal code duplication

- rename family specific resid_dev to have leading underscore
- change `resid_dev` in family superclass to handle the generic sign sqrt var_weights and scale.
- delete the corresponding part again from GLM resid_deviance
- minimal adjustment to docs (no full templating for now) generic Family.resid_dev should mention that details are in family specific `_resid_dev`

This should make family resid_dev backwards compatible with 0.8 and fix the GEE bug.
